staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -735,16 +735,27 @@ func (s *Controller) nodeSyncInternal(ctx context.Context, workers int) {
 	if !s.needFullSyncAndUnmark() {
 		// The set of nodes in the cluster hasn't changed, but we can retry
 		// updating any services that we failed to update last time around.
-		s.servicesToUpdate = s.updateLoadBalancerHosts(ctx, s.servicesToUpdate, workers)
+		// Besides, only latest Service specs should be applied.
```

Comment:
This comment is not very useful IMO, we should add some comments about why we're doing a cache get of every Service again here. 
```
// The set of nodes in the cluster hasn't changed, but we can retry
// updating any services that we failed to update last time around.
// It is required to call `s.cache.get()` on each Service in case there was
// an update event that occured between retries.
````

Reply:
Thank you! Updated.